Changing imports removes comment at the top of a file

Given a Go source file with a comment at the top of the file (not a package comment):

```Go
// Copyright 2022 ...

package main

import (
	"cmd/internal/edit"
)

func main() {
	_ = edit.Buffer
}
```

and a patch to modify the imports:

```Patch
@@
var randomIdentifier identifier
@@
-import "cmd/internal/edit"
+import "github.com/foo/bar/internal/edit"

edit.randomIdentifier
```

the comment at the top is removed by gopatch, resulting with the following file:

```Go
package main

import "github.com/foo/bar/internal/edit"

func main() {
	_ = edit.Buffer
}
```

The respective `testdata` file:

```none
-- in.patch --
@@
var randomIdentifier identifier
@@
-import "cmd/internal/edit"
+import "github.com/foo/bar/internal/edit"

edit.randomIdentifier

-- comment.in.go --
// Copyright 2022 ...

package main

import (
	"cmd/internal/edit"
)

func main() {
	_ = edit.Buffer
}

-- comment.out.go --
// Copyright 2022 ...

package main

import "github.com/foo/bar/internal/edit"

func main() {
	_ = edit.Buffer
}
```

This returns:

```shell
$ go test .
--- FAIL: TestIntegration (0.04s)
    --- FAIL: TestIntegration/replace_import_with_top_level_comment (0.00s)
        --- FAIL: TestIntegration/replace_import_with_top_level_comment/comment (0.00s)
            e2e_test.go:127: 
                        Error Trace:    e2e_test.go:127
                        Error:          Not equal: 
                                        expected: "// Copyright 2022 ...\n\npackage main\n\nimport \"github.com/foo/bar/internal/edit\"\n\nfunc main() {\n\t_ = edit.Buffer\n}\n"
                                        actual  : "package main\n\nimport \"github.com/foo/bar/internal/edit\"\n\nfunc main() {\n\t_ = edit.Buffer\n}\n"
                                    
                                        Diff:
                                        --- Expected
                                        +++ Actual
                                        @@ -1,3 +1 @@
                                        -// Copyright 2022 ...
                                        -
                                         package main
                        Test:           TestIntegration/replace_import_with_top_level_comment/comment
FAIL
FAIL    github.com/uber-go/gopatch      0.047s
FAIL
```

--- a/main.go
+++ b/main.go
@@ -325,6 +325,10 @@ func (r *patchRunner) Apply(filename string, f *ast.File) (*ast.File, bool) {
 func cleanupFilePos(tfile *token.File, cl engine.Changelog, comments []*ast.CommentGroup) {
 	linesToDelete := make(map[int]struct{})
 	for _, dr := range cl.ChangedIntervals() {
+		if dr.Start == token.NoPos {
+			continue
+		}
+
 		for i := tfile.Line(dr.Start); i < tfile.Line(dr.End); i++ {
 			if i > 0 {
 				linesToDelete[i] = struct{}{}

--- a/testdata/replace_import_with_top_level_comment
+++ b/testdata/replace_import_with_top_level_comment
@@ -1,0 +1,32 @@
+-- in.patch --
+@@
+var randomIdentifier identifier
+@@
+-import "cmd/internal/edit"
++import "github.com/foo/bar/internal/edit"
+
+edit.randomIdentifier
+
+-- comment.in.go --
+// Copyright 2022 ...
+
+package main
+
+import (
+	"cmd/internal/edit"
+)
+
+func main() {
+	_ = edit.Buffer
+}
+
+-- comment.out.go --
+// Copyright 2022 ...
+
+package main
+
+import "github.com/foo/bar/internal/edit"
+
+func main() {
+	_ = edit.Buffer
+}

